File: Ch14/01.hello_single/hello_single.rs
use std::io::{stdout, Write};
use std::thread::sleep;
use std::time::Duration;
const NUM: usize =  5;

fn main() {
    print_msg("hello");
    print_msg("world\n");
}

fn print_msg(m: &str) {
    for _ in 0..NUM {
        print!("{}", m);
        stdout().flush().unwrap(); 
        sleep(Duration::from_secs(1)); 
    }
}

File: Ch14/01.hello_single/hello_single.c
/*
 * hello_single.c: a single threaded hello world program
*/

#include <stdio.h>
#include <unistd.h>
#define NUM 5

void print_msg(char *);

int main(void)
{
	print_msg("hello");
	print_msg("world\n");
}

void print_msg(char *msg)
{
	for (int i = 0; i < NUM; i += 1) {
		printf("%s", msg);
		fflush(stdout);
		sleep(1);
	}
}
